### PR TITLE
Fixed crash in "checking mods" mode (fixes #31938)

### LINKED ISSUE #31938: --check-mods SIGSEGV
# Describe the bug

crashes when run with --check-mods argument

# Steps To Reproduce

1. `./cataclysm --check-mods`

# Expected behavior
No crash
# Versions and configuration
 - OS: Ubuntu 19.04
 - Game Version: 0.D-5055-g57b8e63, 0.D-5062-g7f06d55
 - Graphics version: Tiles and Terminal

# Additional context

With no worlds or empty world

[crash.log](https://github.com/CleverRaven/Cataclysm-DDA/files/3339883/crash.log)
[debug.log](https://github.com/CleverRaven/Cataclysm-DDA/files/3339884/debug.log)

### PR DESCRIPTION
**Pull request by ipcyborg**
#### Summary
SUMMARY: Bugfixes "'--check-mods SIGSEGV' Fix crash when run with '--check-mods' argument."

#### Purpose of change
Fixes #31938

#### Describe the solution
The original issue is in `get_world_base_save_path`, if `world_generator->active_world` is null. Because there is no active world yet. Added a check here. 

Then it still crashed on the second mod check. So, I have to ensure that there is a clean world created and activated before checking every mod.

#### Describe alternatives you've considered
Tried to use debugmsg in `get_world_base_save_path`, but failed. Don't know why, but will still crash. Probably need to check every usage of `get_world_base_save_path` and ensure that the world is already created.

#### Additional context
It is not very obvious how to create and cleanly delete a world, and then repeat a process again. 
For example, if I don't use these lines:
`MAPBUFFER.reset();`
`overmap_buffer.clear();`
then it will still be a crash on second mod check.

So, I'm not really sure that I have done this correctly.

Also, after complete mod check there are some errors and warnings in some mods. But this is out of this PR scope.

### CODE REVIEW

File: src/game.cpp [hunk 1]
```diff
@@ -357,8 +375,13 @@ bool game::check_mod_data( const std::vector<mod_id> &opts, loading_ui &ui )
         } catch( const std::exception &err ) {
             std::cerr << "Error loading data: " << err.what() << std::endl;
         }
-    }
 
+        auto world_name = world_generator->active_world->world_name;
```
Comment (anothersimulacrum) [inline src/game.cpp]:
```suggestion
        std::string world_name = world_generator->active_world->world_name;
```

File: src/game.cpp [hunk 2]
```diff
@@ -318,16 +318,34 @@ bool game::check_mod_data( const std::vector<mod_id> &opts, loading_ui &ui )
...
+
         // if no loadable mods then test core data only
         try {
             load_core_data( ui );
             DynamicDataLoader::get_instance().finalize_loaded_data( ui );
         } catch( const std::exception &err ) {
             std::cerr << "Error loading data from json: " << err.what() << std::endl;
         }
+
+        auto world_name = world_generator->active_world->world_name;
```
Comment (anothersimulacrum) [inline src/game.cpp]:
```suggestion
        std::string world_name = world_generator->active_world->world_name;
```

**Comment by AMurkin** [comment]
> --check-mods SIGSEGV

That's not really a changelog entry imo. Can you decrypt it?

**Comment by ipcyborg** [comment]
Fixed title and summary. Thanks!
Sorry about it, just copied the title from the issue description.

**Comment by anothersimulacrum** [comment]
As of right now, that's not a valid summary, I'd suggest
`SUMMARY: Bugfixes "'--check-mods SIGSEGV' Fix crash when run with '--check-mods' argument.`

**Pull request merged**